src/lambda/handler-runner/docker-runner/DockerContainer.js

```diff
@@ -41,52 +45,78 @@ export default class DockerContainer {
-      `${codeDir}:/var/task:ro,delegated`,
-      '-p',
-      `${port}:9001`,
-      '-e',
-      'DOCKER_LAMBDA_STAY_OPEN=1', // API mode
-    ]
 
+    const envs = ['DOCKER_LAMBDA_STAY_OPEN=1']
     entries(this.#env).forEach(([key, value]) => {
-      dockerArgs.push('-e', `${key}=${value}`)
+      envs.push(`${key}=${value}`)
     })
 
+    const dockerArgs = {
+      Image: this.#imageNameTag,
+      Env: envs,
+      ExposedPorts: {
+        '9001/tcp': { HostPort: `${port}` },
+      },
+      HostConfig: {
+        PortBindings: { '9001/tcp': [{ HostPort: `${port}` }] },
+        Binds: [`${codeDir}:/var/task:ro,delegated`],
+      },
+      AttachStdin: false,
+      AttachStdout: true,
+      AttachStderr: true,
+    }
+
     if (platform() === 'linux') {
       // Add `host.docker.internal` DNS name to access host from inside the container
       // https://github.com/docker/for-linux/issues/264
       const gatewayIp = await this._getBridgeGatewayIp()
-      dockerArgs.push('--add-host', `host.docker.internal:${gatewayIp}`)
+      dockerArgs.HostConfig.ExtraHosts = [`host.docker.internal:${gatewayIp}`]
     }
 
-    const { stdout: containerId } = await execa('docker', [
-      'create',
-      ...dockerArgs,
-      this.#imageNameTag,
-      this.#handler,
-    ])
-
-    const dockerStart = execa('docker', ['start', '-a', containerId], {
-      all: true,
-    })
-
-    await new Promise((resolve, reject) => {
-      dockerStart.all.on('data', (data) => {
-        const str = data.toString()
-        console.log(str)
-        if (str.includes('Lambda API listening on port')) {
-          resolve()
-        }
+    const container = await this.#dockerClient.createContainer(dockerArgs)
+
+    const containerLogs = (runningContainer) => {
+      return new Promise((resolve, reject) => {
+        // create a single stream for stdin and stdout
+        const logStream = new stream.PassThrough()
+
+        logStream.on('data', (chunk) => {
+          if (chunk.toString('utf8').includes('Lambda API listening on port')) {
+            resolve()
+          }
+        })
+
+        runningContainer.logs(
+          {
+            follow: true,
+            stdout: true,
+            stderr: true,
+          },
+          (err, strm) => {
+            if (err) {
+              reject(err)
+              return console.error(err.message)
+            }
+            runningContainer.modem.demuxStream(strm, logStream, logStream)
+            strm.on('end', () => {
+              logStream.end()
+              resolve()
+            })
+
+            return setTimeout(() => {
+              strm.destroy()
+              reject(new Error('log timeout'))
+            }, 3000)
+          },
+        )
       })
+    }
 
-      dockerStart.on('error', (err) => {
-        reject(err)
-      })
-    })
+    await container.start()
+
+    await containerLogs(container)
 
-    this.#containerId = containerId
+    this.#containerId = container.id
```

Comment:
I think `DockerContainer` class no longer needs to keep container id.
It would be better to set a container directly.
```suggestion
    this.#container = container
```